api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/ClusterEndpointProvider.java

```diff
@@ -5,6 +5,44 @@
  */
 package io.kroxylicious.proxy.service;
 
+import java.util.Optional;
+
 public interface ClusterEndpointProvider {
+
+    /** Address of the cluster's bootstrap address, includes the port.
+     *
+     * @return cluster's bootstrap address in the form host:port
+     */
     String getClusterBootstrapAddress();
+
+    /** Address of broker with the given node id, includes the port. Note that
+     * {@code nodeId} are consecutively numbered and  begin at zero.
```

Comment:
Wondering of this business rule is too strict.  I suppose that a cluster's broker topology may evolve over time.  `nodeIds` might become unused owing to say hardware failure.